Gruntfile.js

```diff
@@ -39,101 +66,122 @@ module.exports = function( grunt ) {
+          "public/friendlycode/vendor/slowparse/test/node/qunit-shim.js"
         ]
+      },
+      frontend: {
+        files: {
+          src: [
+            "./frontend/scripts/**/*.js",
+            "./frontend/scripts/*.js"
+          ]
+        }
       }
     },
-    lesslint: {
-      src: ["public/stylesheets/userbar.less"],
+
+    // Minification
+    uglify: {
+      default: {
+        files: {
+          'public/scripts/thimble.min.js': ['public/scripts/thimble.js']
+        }
+      }
+    },
+    cssmin: {
+      default: {
+        files: [{
+          expand: true,
+          cwd: 'frontend/styles',
+          src: ['*.css'],
+          dest: 'public/styles',
+          ext: '.min.css'
+        }]
+      }
+    },
+
+    // Build
+    browserify: {
+      default: {
+        src: "./frontend/scripts/index.js",
+        dest: "./public/scripts/thimble.js",
         options: {
-          csslint: {
-            "adjoining-classes": false,
-            "box-model": false,
-            "box-sizing": false,
-            "bulletproof-font-face": false,
-            "compatible-vendor-prefixes": false,
-            "floats": false,
-            "font-sizes": false,
-            "ids": false,
-            "important": false,
-            "outline-none": false,
-            "overqualified-elements": false,
-            "qualified-headings": false,
-            "regex-selectors": false,
-            "star-property-hack": false,
-            "underscore-property-hack": false,
-            "universal-selector": false,
-            "unique-headings": false,
-            "unqualified-attributes": false,
-            "vendor-prefix": false,
-            "zero-units": false
+          alias: {
+            // Specify bower dependencies here for use with commonjs
+            // requires e.g. require('zepto');
+            "zepto": "./bower_components/zepto/zepto.min.js"
+          },
+          browserifyOptions: {
+            commondir: false
           }
         }
-    },
-    jshint: {
-      options: {
-        "-W069": true // ignore "['...'] is better written in dot notation." warnings
       },
-      files: [
-        "Gruntfile.js",
-        "app.js",
-        "lib/**/*.js",
-        "routes/**/*.js",
-        "public/friendlycode/js/**/*.js",
-        "public/friendlycode/vendor/slowparse/slowparse.js",
-        "public/friendlycode/vendor/slowparse/test/nodetest.js",
-        "public/friendlycode/vendor/slowparse/test/test-slowparse.js",
-        "public/friendlycode/vendor/slowparse/test/node/qunit-shim.js"
-      ]
     },
-    execute: {
-      target: {
-        src: [
-          "public/friendlycode/vendor/slowparse/test.js"
-        ]
+    less: {
+      default: {
+        options: {
+          compress: true
+        },
+        files: {
+          "frontend/styles/error.css": "frontend/styles/error.less"
+        }
       }
     },
 
-    // Bramble tasks
+    // Workflow
     'npm-checkBranch': {
```

Comment:
Do we still need all this for Thimble?  We shouldn't really need submodules and such going forward, right?